Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -2344,12 +2350,72 @@ def get_ioc_value(ioc, id_to_obj):
     @staticmethod
     def extract_ioc_value(ioc_obj, key: str = "name"):
         """
-        Extract SHA-256 from specific key, default key is name.
-        "([file:name = 'blabla' OR file:name = 'blabla'] AND [file:hashes.'SHA-256' = '1111'])" -> 1111
+        Extract indicator values from a specific key; the default key is name.
+        Extracts values for all supported indicator types from STIX patterns.
+
+        Examples:
+        - "[file:hashes.'SHA-256' = '1111']" -> "1111" (SHA-256 hash)
+        - "[ipv4-addr:value = '1.2.3.4']" -> "1.2.3.4" (IP address)
+        - "[domain-name:value = 'example.com']" -> "example.com" (domain)
+        - "[windows-registry-key:key = 'HKEY_LOCAL_MACHINE\\Software\\Malware']" -> "HKEY_LOCAL_MACHINE\\Software\\Malware"
+        - "[windows-registry-key:value = 'MalwareValue']" -> "MalwareValue"
+
+        Returns the first found indicator value based on a priority order:
+        1. file:hashes.'SHA-256'
+        2. ipv4-addr:value
+        3. domain-name:value
+        4. url:value
+        5. Other supported indicator types defined in STIX_SUPPORTED_TYPES
+
+        If multiple indicators of the same type are found, returns the first one.
```

Comment:
I remember it is by design but is it documented in a place visible to the customer?